pkg/cmd/workspace/logs.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2024 Daytona Platforms Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package workspace
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/daytonaio/daytona/cmd/daytona/config"
+	"github.com/daytonaio/daytona/internal/util"
+	apiclient_util "github.com/daytonaio/daytona/internal/util/apiclient"
+	logs_view "github.com/daytonaio/daytona/pkg/views/logs"
+	"github.com/daytonaio/daytona/pkg/views/workspace/selection"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/spf13/cobra"
+)
+
+var followFlag bool
+var retryFlag bool
+
+var LogsCmd = &cobra.Command{
+	Use:     "logs [WORKSPACE] [PROJECT]",
+	Short:   "View logs for a workspace or project",
+	GroupID: util.WORKSPACE_GROUP,
+	Args:    cobra.RangeArgs(0, 2),
+	Aliases: []string{"log"},
+	Run: func(cmd *cobra.Command, args []string) {
+		c, err := config.GetConfig()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		activeProfile, err := c.GetActiveProfile()
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		query := ""
+		if retryFlag && followFlag {
```

Comment:
![Screenshot 2024-08-19 at 15 47 51](https://github.com/user-attachments/assets/864a5f2d-e31d-4837-9cac-178e9924b95c)

There is a difference in how the `--follow` and base commands behave.
As you can see in the screenshot, the non-follow command displays only 2 lines (this is also not consistent, sometimes it displays 3 lines) while the follow command displays all 4 lines. 
The expected behavior is that both commands display the 4 lines because they are present in the log when invoking the command.

Reply:
fixed the behaviour

Reply:
I imagine it has something to do with the way the logs are read on the backend.

Please look into that. You can first check if the issue occurs without your modification to the read process in `internal/util/log_reader.go`